kola/harness.go

```diff
@@ -418,16 +418,30 @@ func RunTest(t *register.Test, pltfrm string) error {
 		}
 	}
 
-	// run test
-	err = t.Run(tcluster)
-
 	// give some time for the remote journal to be flushed so it can be read
 	// before we run the deferred machine destruction
-	if err != nil {
-		time.Sleep(10 * time.Second)
-	}
+	defer func() {
+		if err != nil {
+			time.Sleep(2 * time.Second)
+		}
+	}()
+
+	defer func() {
+		if r := recover(); r != nil {
+			switch r := r.(type) {
+			// can't have multiple types in a single case.
+			case error:
+				err = r
+			case skip.Skip:
```

Comment:
Yeah, the spec doesn't appear to specify if type switches are evaluated in-order like expression switches, and their example doesn't try to match interface types either. Though it seems fair to indicate our intended behavior by keeping the order of evaluation we want in the code.

Also you can reduce the nesting by putting nil as the first case of the switch instead of inside an if